src/pubsub/ua_pubsub_config.c

```diff
@@ -0,0 +1,1693 @@
+        binFile = (UA_UABinaryFileDataType*)src->content.decoded.data;
+    } else {
+        UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, "[UA_getPubSubConfig] Reading extensionObject failed");
+        return UA_STATUSCODE_BADINVALIDARGUMENT;
+    }
+
+    if(binFile->body.arrayLength != 0 || binFile->body.arrayDimensionsSize != 0) {
+        UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, 
+                     "[UA_getPubSubConfig] Loading multiple configurations is not supported");
+        return UA_STATUSCODE_BADNOTIMPLEMENTED;
+    }
+
+    if(UA_NodeId_equal(&binFile->body.type->typeId, &UA_TYPES[UA_TYPES_PUBSUBCONFIGURATIONDATATYPE].typeId)) {
+        *dst = (UA_PubSubConfigurationDataType*)binFile->body.data;
+    } else {
+        UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, 
+                     "[UA_getPubSubConfig] Invalid datatype encoded in the binary file");
+        return UA_STATUSCODE_BADTYPEMISMATCH;
+    }
+    return UA_STATUSCODE_GOOD;
+}
+
+/* UA_updatePubSubConfig() */
+/**
+ *  @brief      Configures a PubSub Server with given PubSubConfigurationDataType object
+ * 
+ *  @param      server                  [bi]    Server object
+ *  @param      configurationParameters [in]    PubSub Configuration parameters
+ * 
+ *  @return     UA_StatusCode
+ */
+UA_StatusCode 
+UA_updatePubSubConfig(UA_Server* server, const UA_PubSubConfigurationDataType *const configurationParameters)
+{
+    if(server == NULL || configurationParameters == NULL) {
+        UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, "[UA_updatePubSubConfig] Invalid argument");
+        return UA_STATUSCODE_BADINVALIDARGUMENT;
+    }
+    
+    UA_PubSubManager_delete(server, &(server->pubSubManager));
+        
+    /* Configuration of Published DataSets: */
+    UA_UInt32 pdsCount = (UA_UInt32)configurationParameters->publishedDataSetsSize;
+    UA_NodeId *publishedDataSetIdent = (UA_NodeId*)UA_calloc(pdsCount, sizeof(UA_NodeId));
+    if(publishedDataSetIdent == NULL) {
+        return UA_STATUSCODE_BADOUTOFMEMORY;
+    }
+    UA_StatusCode statusCode = UA_STATUSCODE_GOOD;
+    for(UA_UInt32 pdsIndex = 0; pdsIndex < pdsCount; pdsIndex++) {
+        statusCode = UA_createPublishedDataSet(server, &configurationParameters->publishedDataSets[pdsIndex], 
+                                               &publishedDataSetIdent[pdsIndex]);
+        if(statusCode != UA_STATUSCODE_GOOD) {
+            UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, "[UA_updatePubSubConfig] PDS creation failed");
+            free(publishedDataSetIdent);
+            return statusCode;
+        }
+    }
+
+    /* Configuration of PubSub Connections: */
+    if(configurationParameters->connectionsSize < 1) {
+        UA_LOG_WARNING(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, 
+                       "[UA_updatePubSubConfig] no connection in UA_PubSubConfigurationDataType");
+        free(publishedDataSetIdent);
```

Comment:
Always use the open62541 macros to achieve the maximum portability e.g. UA_free. I will not mention this anymore.